homeassistant/components/tplink/light.py

```diff
@@ -101,6 +104,11 @@ async def async_turn_on(self, **kwargs: Any) -> None:
             await self.device.set_hsv(hue, sat, brightness, transition=transition)
             return
 
+        if ATTR_EFFECT in kwargs:
+            assert isinstance(self.device, SmartLightStrip)
+            await self.device.set_effect(kwargs[ATTR_EFFECT])
```

Comment:
I think we need to figure out what's the best approach to fix these mypy errors, the version bump to 0.4.1 has been pending for a while due to the inconsistent API (#64123).

Reply:
I made another class for the light strips.  This should be the only assert left because separating turn_on was more complex than leaving the assert here.